db.go

```diff
@@ -0,0 +1,73 @@
+package auth
+
+import (
+	"errors"
+
+	"github.com/jinzhu/gorm"
+	_ "github.com/jinzhu/gorm/dialects/postgres"
+)
+
+const DefaultMaxIdleConns = 5
+
+var ErrorNoConnectionString = errors.New("A connection string must be specified on the first call to Get")
+
+// DatabaseHandler abstracts away common persistence operations needed for this package
+type DatabaseHandler interface {
```

Comment:
may want to include a `close` on the db handler

Reply:
good thought...may add in a `Close()` to the `DatabaseHandler` but not sure I want to bog down the `Authenticator` api with having a `Close`... another option is to make the `auth` struct's `databaseHandler` publically accessible. Either way, I'll defer this for future discussion and thought. For a long running application like this with pooled db connections I think we should be okay in the interim but this deserves revisiting to handle in the proper way.

Reply:
may just make sense to throw in an Authenticator.Close() that callers can `defer`